### PR TITLE
feat(community-updates): improve filters with backend integration

### PR DESCRIPTION
**Pull request by amaurymagalhaes**
## Summary
- Move all filtering logic to backend (status, programId, projectId)
- Add loading states to Program and Project filter dropdowns
- Use project slug instead of UID in URL parameters for better readability
- Support both UID and slug for project filter lookup

## Changes
- **Updates Page**: Integrated backend filtering, added program/project filter dropdowns
- **SearchWithValueDropdown**: Added `isLoading` prop with spinner and disabled state
- **useCommunityProjects**: Fixed slug extraction from API response
- **useCommunityProjectUpdates**: Updated to pass filter params to backend
- **Tests**: Added tests for query keys and milestone card component

## Test plan
- [x] Status filter (all, pending, completed, past_due) works correctly
- [x] Program filter shows loading state while fetching
- [x] Project filter shows loading state while fetching
- [x] Selecting project uses slug in URL (e.g., `projectId=cidgravity-filecoin-gateway`)
- [x] Filters work together (status + program + project)
- [x] Page handles direct URL with slug parameter correctly

## Summary by CodeRabbit

* **New Features**
  * Program and project filters on community updates with URL state sync; pagination resets when filters change.
  * Added "Past Due" status filter.
  * Search dropdowns show a loading state with spinner and disable interaction while loading.

* **Bug Fixes**
  * Grant links now navigate to funding pages when present and gracefully fall back to project paths when slugs/details are missing.
  * Project titles render as bold plain text; milestone title styling and "view" links adapt to grant presence.

* **Tests**
  * Expanded coverage for filtering, linking, slug/UID fallbacks, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: app/community/[communityId]/updates/page.tsx [hunk 1]
```diff
@@ -119,54 +199,113 @@ export default function CommunityUpdatesPage() {
...
-          <div className="flex items-center gap-2">
-            <span className="text-sm text-gray-600 dark:text-gray-400">
+        {/* Filter row */}
+        <div className="px-3 py-4 bg-gray-100 dark:bg-zinc-900 rounded-lg flex flex-row items-center w-full gap-8 max-lg:flex-col max-lg:gap-4 max-lg:justify-start max-lg:items-start">
+          {/* Left side - Filters */}
+          <div className="flex flex-row items-center gap-6 flex-wrap max-lg:w-full">
+            {/* Status filter dropdown */}
+            <Listbox value={selectedFilter} onChange={handleFilterChange}>
+              <div className="relative">
+                <Listbox.Button className="flex items-center gap-2 px-4 py-3 text-sm font-medium text-gray-700 bg-white border border-gray-300 rounded-lg hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-brand-blue dark:bg-zinc-800 dark:text-zinc-200 dark:border-zinc-600 min-w-[120px]">
```
Comment (brunodmsi) [inline app/community/[communityId]/updates/page.tsx]: add aria-label attrs (wcag)

**Review by brunodmsi — APPROVED** [review]
